server/homebrew.api.js

```diff
@@ -6,13 +6,52 @@ const GoogleActions = require('./googleActions.js');
 const Markdown = require('../shared/naturalcrit/markdown.js');
 const yaml = require('js-yaml');
 const asyncHandler = require('express-async-handler');
+const { nanoid } = require('nanoid');
 
 // const getTopBrews = (cb) => {
 // 	HomebrewModel.find().sort({ views: -1 }).limit(5).exec(function(err, brews) {
 // 		cb(brews);
 // 	});
 // };
 
+const getBrew = (accessType)=>{
+	return async (req, res, next)=>{
+		const { brew } = req;
+
+		if(!brew) {
+			let id = req.params.id, googleId = req.body?.googleId;
+
+			if(id.length > 12) {
+				googleId = id.slice(0, -12);
+				id = id.slice(-12);
+			}
+			let found = await HomebrewModel.get(accessType === 'edit' ? { editId: id } : { shareId: id })
+				.catch((err)=>{
+					if(googleId) {
+						console.warn(`Unable to find document stub for ${accessType}Id ${id}`);
+					} else {
+						console.warn(err);
+					}
+				});
+
+			if(googleId || found?.googleId) {
+				const googleBrew = await GoogleActions.getGoogleBrew(googleId || found?.googleId, id, accessType)
+					.catch((err)=>{
+						console.warn(err);
+					});
+				if(!found && !googleBrew) throw 'Brew not found in database or Google Drive';
+				found = found ? Object.assign(excludeStubProps(found), excludeGoogleProps(googleBrew)) : googleBrew;
+			} else if(!found) {
+				throw 'Brew not found in database';
+			}
```

Comment:
Instead of `else if`, does this work if we instead treat it as a catch-all for any missing brew (including from database)?

```suggestion
			}
			
			if(!found) {
				throw 'Brew not found';
			}
```

In essence, if we just cleanly divide this up into 4 segments:

1) Convert IDs to Google IDs as needed
2) Check for a stub (or full brew) -> No stub: Warn: "Brew not found in Database" or "Stub not found for Google Brew"
3) If Google ID, check for Google Brew -> No Google Brew: Warn: "Brew not found in Google Drive"
   1) Merge Google Brew into Stub if needed
4) If no brew, throw error

Reply:
Yeah that's a solid plan